### ISSUE TITLE
Group patch/minor and lockFileMaintenance

### ISSUE THREAD

**Issue by tpansino** [renovatebot/renovate#13470]
### What would you like Renovate to be able to do?

As a user
I would like to be able to group patch/minor version bumps and lock file maintenance into a single PR
So that I can update all of my dependencies weekly with a single merge to each of my GitHub repos.

Multiple PRs can be time-consuming to ship if CI/CD is long-running. It seems like it should be feasible to group these update types.

### If you have any ideas on how this should be implemented, please tell us here.

See #13355

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
This behaviour is different per manager too

**Comment by tpansino** [comment]
Poetry, Terraform lock, NPM, are my wishlist, in that order. 😄

Is there any further information I can provide to help guide the implementation?

**Comment by rarkins** [comment]
Yes, minimal reproductions of each which can be tested against. e.g. 1-2 outdated dependencies, or whatever is the minimum config/code which reproduces the problem. See the link above for more details

**Comment by PeterNitscheMI** [comment]
Hi @rarkins, me and my team are really interested in the group support of lockFileMaintenance for **npm packages** with yarn.lock.
As requested, I have created a minimal reproduction repository (incl. readme.md):
https://github.com/PeterNitscheMI/RenovateLFM

**Comment by rarkins** [comment]
Reproduction repo forked to https://github.com/renovate-reproductions/13470

I'm not sure that "grouping" lock file maintenance is the right way here. For example I can see a use case where someone not doing grouping who wants every PR to have a fully refreshed lock file each time. In this case a new option of e.g. "reuseLockFile" true/false (Defaulting to true) might be better.

**Comment by PeterNitscheMI** [comment]
Sounds good! It seems your proposed solution would fulfill the needs of @tpansino, me and the others voting with 👍 
Thanks for caring!

**Comment by rarkins** [comment]
OK, in this case it would be:
- new option `lockFileRegenerate` which defaults to `false`
- option will be passed to artifacts updating functions in managers
- if managers support this option then it means they'll not reuse the existing lock file and instead regenerate it with latest versions (i.e. updating every possible dependency in the lock file within the constraints of the package file)

This can begin with npm/yarn/pnpm. PRs welcome

**Comment by palfrey** [comment]
I believe this is done now?

**Comment by rarkins** [comment]
No
